### ISSUE TITLE
Update erorr-prone to 2.3.1

### ISSUE THREAD

**Issue by jyane** [grpc/grpc-java#4452]
https://github.com/google/error-prone/releases

**Comment by jyane** [comment]
@ejona86 Are there any concerns?

**Comment by carl-mastrangelo** [comment]
I believe this drops support for DoNotMock, which we use.

**Comment by jyane** [comment]
I didn't noticed that drops support for DoNotMock...
https://github.com/google/error-prone/pull/989
I don't know why it was removed, it seems to be internal problems.

**Comment by ejona86** [comment]
#4659 removed the `@DoNotMock`s, so this is now possible.

**Comment by ejona86** [comment]
ErrorProne was upgraded to 2.3.2 in 183e1f67

**Comment by vinodhabib** [comment]
> 183e1f6

Yes Eric, This issue is related to an error-prone upgrade to 2.3.1 and as per the release documentation (https://github.com/google/error-prone/releases?q=&expanded=true ) it's already upgraded to 2.36.0, so no action required
